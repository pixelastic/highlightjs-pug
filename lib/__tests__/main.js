const current = require('../pug.js');
const highlightjs = require('highlightjs');
highlightjs.registerLanguage('pug', current);

// !!! XML
// %html
//   %body
//     %h1.jumbo{:id=>"a", :style=>'font-weight: normal', :title=>title} highlight.js
//     /html comment
//     -# ignore this line
//     %ul(style='margin: 0')
//     -items.each do |i|
//       %i= i
//     = variable
//     =variable2
//     ~ variable3
//     ~variable4
//     The current year is #{DataTime.now.year}.

describe('current', () => {
  it('should highlight html tags', async () => {
    const input = dedent`
    p Text
    `;
    const actual = highlightjs.highlight('pug', input);
    expect(actual.value).toMatchSnapshot();
  });
  it('should highlight html classes', async () => {
    const input = dedent`
    .bg-red Text
    `;
    const actual = highlightjs.highlight('pug', input);
    expect(actual.value).toMatchSnapshot();
  });
  it('should highlight id classes', async () => {
    const input = dedent`
    #uniqueId Text
    `;
    const actual = highlightjs.highlight('pug', input).value;
    expect(actual).toMatchSnapshot();
  });
  it('tag with class', async () => {
    const input = dedent`
    p.bg-red Text
    `;
    const actual = highlightjs.highlight('pug', input).value;
    expect(actual).toMatchSnapshot();
  });
});
