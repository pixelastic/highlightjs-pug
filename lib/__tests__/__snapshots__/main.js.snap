// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`current should highlight html classes 1`] = `"<span class=\\"hljs-class\\">.bg-red</span> Text"`;

exports[`current should highlight html tags 1`] = `"<span class=\\"hljs-tag\\">p</span> Text"`;

exports[`current should highlight id classes 1`] = `"<span class=\\"hljs-id\\">#uniqueId</span> Text"`;

exports[`current tag with class 1`] = `"<span class=\\"hljs-tag\\">p</span><span class=\\"hljs-class\\">.bg-red</span> Text"`;
